src/py/flwr/cli/run/run.py

```diff
@@ -31,54 +36,85 @@ class Engine(str, Enum):
     SIMULATION = "simulation"
 
 
+# pylint: disable-next=too-many-locals
 def run(
     engine: Annotated[
         Optional[Engine],
         typer.Option(case_sensitive=False, help="The ML framework to use"),
     ] = None,
+    use_superexec: Annotated[
+        bool,
+        typer.Option(
+            case_sensitive=False, help="Use this flag to use the new SuperExec API"
+        ),
+    ] = False,
 ) -> None:
     """Run Flower project."""
-    typer.secho("Loading project configuration... ", fg=typer.colors.BLUE)
-
-    config, errors, warnings = config_utils.load_and_validate()
-
-    if config is None:
-        typer.secho(
-            "Project configuration could not be loaded.\npyproject.toml is invalid:\n"
-            + "\n".join([f"- {line}" for line in errors]),
-            fg=typer.colors.RED,
-            bold=True,
-        )
-        sys.exit()
-
-    if warnings:
-        typer.secho(
-            "Project configuration is missing the following "
-            "recommended properties:\n" + "\n".join([f"- {line}" for line in warnings]),
-            fg=typer.colors.RED,
-            bold=True,
+    if use_superexec:
+
+        def on_channel_state_change(channel_connectivity: str) -> None:
+            """Log channel connectivity."""
+            log(DEBUG, channel_connectivity)
+
+        channel = create_channel(
+            server_address="127.0.0.1:9093",
+            insecure=True,
+            root_certificates=None,
+            max_message_length=GRPC_MAX_MESSAGE_LENGTH,
+            interceptors=None,
```

Comment:
Instead of indenting the existing code because if the if/else, can we move the new code into a function and start with `if user_superexec`, followed by a call to that function followed by `return`?